### PR TITLE
Marking translatable strings from several frontend components

### PR DESCRIPTION
**Pull request by joolean**

### CODE REVIEW

File: hsreplaynet/static/scripts/src/components/NoDecksMessage.tsx [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
-export default class NoDecksMessage extends React.Component {
+class NoDecksMessage extends React.Component {
 	public render(): React.ReactNode {
 		return (
-			<div className="message-wrapper">
-				<h2>No deck found</h2>
-				{this.props.children}
-			</div>
+			<Trans>
+				<div className="message-wrapper">
```
Comment (jleclanche) [inline hsreplaynet/static/scripts/src/components/NoDecksMessage.tsx]: You shouldn't use Trans here, just t()

File: hsreplaynet/static/scripts/src/components/home/ReplayFeed.tsx [hunk 2]
```diff
@@ -159,44 +160,53 @@ class ReplayFeed extends React.Component<Props, State> {
...
-						href="https://hsdecktracker.net/"
-						target="_blank"
-					>
-						Become a Contributor
-					</a>
-				</div>
-			</div>
+			<>
+				{/* prettier-ignore */}
+				<Trans>
```
Comment (joolean) [inline hsreplaynet/static/scripts/src/components/home/ReplayFeed.tsx]: This look okay?

File: hsreplaynet/static/scripts/src/components/gamehistory/GameHistoryTable.tsx [hunk 3]
```diff
@@ -47,32 +48,36 @@ export default class GameHistoryTable extends React.Component<Props> {
...
-						<div className="match-table-cell hide-below-1600">
-							Duration
-						</div>
-						<div className="match-table-cell hide-below-768">
-							Turns
-						</div>
-						<div className="match-table-cell hide-below-500">
-							Played
-						</div>
+						<Trans>
```
Comment (jleclanche) [inline hsreplaynet/static/scripts/src/components/gamehistory/GameHistoryTable.tsx]: Same

File: hsreplaynet/static/scripts/src/components/home/ReplayFeed.tsx [hunk 4]
```diff
@@ -159,44 +160,53 @@ class ReplayFeed extends React.Component<Props, State> {
...
-					<a
-						className="btn promo-button blue-style"
-						href="https://hsdecktracker.net/"
-						target="_blank"
-					>
-						Become a Contributor
-					</a>
-				</div>
-			</div>
+			<>
```
Comment (jleclanche) [inline hsreplaynet/static/scripts/src/components/home/ReplayFeed.tsx]: Same here

File: hsreplaynet/static/scripts/src/components/metaoverview/OtherArchetype.tsx [hunk 5]
```diff
@@ -7,25 +8,29 @@ interface Props {
...
 				content={
-					<p>
-						This is a collection of all{" "}
-						<PrettyCardClass cardClass={playerClass} />
-						decks that do not fit into one of the popular
-						archetypes.
-						<br />
-						<br />No archetype details are available.
-					</p>
+					<Trans>
```
Comment (jleclanche) [inline hsreplaynet/static/scripts/src/components/metaoverview/OtherArchetype.tsx]: Trans components should always have the fewest tags possible so as to limit the amount of noise for translators. Put it inside the <p>

File: hsreplaynet/static/scripts/src/components/payments/CheckoutForm.tsx [hunk 6]
```diff
@@ -82,20 +83,24 @@ export default class CheckoutForm extends React.Component<Props, State> {
 			methods.push({
 				method: PaymentMethod.CREDITCARD,
 				label: (
-					<strong>
-						<span className="glyphicon glyphicon-credit-card" />&nbsp;Credit
-						Card
-					</strong>
+					<Trans>
+						<strong>
+							<span className="glyphicon glyphicon-credit-card" />&nbsp;Credit
```
Comment (jleclanche) [inline hsreplaynet/static/scripts/src/components/payments/CheckoutForm.tsx]: Just use t()

File: hsreplaynet/static/scripts/src/components/payments/CheckoutForm.tsx [hunk 7]
```diff
@@ -123,22 +130,26 @@ export default class CheckoutForm extends React.Component<Props, State> {
...
-					}))}
-					aria-describedby="payment-method"
-					value={this.state.paymentMethod}
-					onChange={paymentMethod => this.setState({ paymentMethod })}
-					disabled={this.state.disabled}
-				/>
-			</div>
+			<Trans>
+				<div style={{ textAlign: "center" }}>
+					<label id="payment-method">Payment method</label>
```
Comment (jleclanche) [inline hsreplaynet/static/scripts/src/components/payments/CheckoutForm.tsx]: Use t()

File: hsreplaynet/static/scripts/src/components/payments/CheckoutForm.tsx [hunk 8]
```diff
@@ -205,23 +216,27 @@ export default class CheckoutForm extends React.Component<Props, State> {
...
-						<a
-							href="https://hsreplay.net/account/billing/"
-							target="_blank"
-						>
-							billing settings
-						</a>{" "}
-						page.
-					</small>
+					{/* prettier-ignore */}
+					<Trans>
```
Comment (jleclanche) [inline hsreplaynet/static/scripts/src/components/payments/CheckoutForm.tsx]: Nest trans deeper

File: hsreplaynet/static/scripts/src/components/premium/PremiumModal.tsx [hunk 9]
```diff
@@ -1,103 +1,122 @@
...
-									href="/premium/"
-									id="premium-modal-learn-more"
-									target="_blank"
-								>
-									Learn more…
-								</a>
-							</>
-						) : null}
-					</p>
+			<>
```
Comment (jleclanche) [inline hsreplaynet/static/scripts/src/components/premium/PremiumModal.tsx]: What's this extra fragment?

Comment (joolean) [reply hsreplaynet/static/scripts/src/components/premium/PremiumModal.tsx]: I needed something above Trans to put the pragma inside.

**Pull request merged**